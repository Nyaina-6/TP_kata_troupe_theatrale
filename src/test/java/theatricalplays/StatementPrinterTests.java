package theatricalplays;

import static org.approvaltests.Approvals.*;

import java.util.List;
import java.util.Map;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

public class StatementPrinterTests {

    @Test
    void exampleStatement() {
        Map<String, Play> plays = Map.of(
            "hamlet",  new TragedyPlay("Hamlet","tragedy"),
            "as-like", new ComedyPlay("As You Like It" ,"comedy"),
            "othello", new TragedyPlay("Othello","tragedy"));
        
        Customer customer = new Customer("Lala", "C02", 47);
    
        Invoice invoice = new Invoice(customer, List.of(
                new Performance("hamlet", 55),
                new Performance("as-like", 55),
                new Performance("othello", 100)));

        StatementPrinter statementPrinter = new StatementPrinter();
        var result = statementPrinter.print(invoice, plays);

        verify(result);
    }

    @Test
    void statementWithNewPlayTypes() {
        Map<String, Play> plays = Map.of(
            "hamlet",  new TragedyPlay("Hamlet","tragedy"),
            "as-like", new ComedyPlay("As You Like It" ,"comedy"),
            "othello", new TragedyPlay("Othello","tragedy"));
                
        
        Customer customer = new Customer("Lala", "C02", 47);
    
        Invoice invoice = new Invoice(customer, List.of(
                new Performance("hamlet", 155),
                new Performance("as-like", 135),
                new Performance("othello", 140)));
        StatementPrinter statementPrinter = new StatementPrinter();
        Assertions.assertDoesNotThrow(() -> {
            statementPrinter.print(invoice, plays);
        });
    }

    @Test
    void testHtmlStatement() {
        Map<String, Play> plays = Map.of(
            "hamlet",  new TragedyPlay("Hamlet","tragedy"),
            "as-like", new ComedyPlay("As You Like It" ,"comedy"),
            "othello", new TragedyPlay("Othello","tragedy"));

        Customer customer = new Customer("Lala", "C02", 47);
    
        Invoice invoice = new Invoice(customer, List.of(
                new Performance("hamlet", 55),
                new Performance("as-like", 55),
                new Performance("othello", 100)));

        StatementPrinter statementPrinter = new StatementPrinter();
        var htmlresult = statementPrinter.toHTML(invoice, plays);

        verify(htmlresult);
    }
}






