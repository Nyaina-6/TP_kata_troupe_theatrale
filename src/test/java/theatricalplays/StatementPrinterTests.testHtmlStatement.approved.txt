<html>
<head><title>Invoice</title></head>
<body>
<h1>Invoice</h1>
<li><ul>Lala</ul></li>
<table border=1 >
<tr><th>Play</th><th>Seats</th><th>Amount</th></tr>
<tr><td>Hamlet</td><td>55</td><td>$650.00</td></tr>
<tr><td>As You Like It</td><td>55</td><td>$740.00</td></tr>
<tr><td>Othello</td><td>100</td><td>$1,100.00</td></tr>
<tr><td  colspan =2 >Total </td><td>$2,490.00</td><tr>
<tr><td  colspan =2 >Discount </td><td> - $15.00</td><tr>
<tr><td  colspan =2 >TOTAL </td><td>$2,475.00</td><tr>
</table>
<p>Your loyality points</p>
<table border=1 >
<tr><th>Credits</th><th>Amount</th></tr>
<tr><td>Earned</td><td>178</td></tr>
<tr><td>Actual</td><td>28</td></tr>
</table>
<p>You must pay within 15 days otherwise your reservation will be cancelled.</p>
</body>
</html>